typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -74,16 +79,30 @@ import Numeric.Natural (Natural)
 -- helpful utility for use in custom drives.
 --
 
-
 --
--- Driver for normal peers
+-- Tracing
 --
 
 -- | Structured 'Tracer' output for 'runPeer' and derivitives.
 --
 data TraceSendRecv ps = TraceSendMsg (AnyMessage ps)
                       | TraceRecvMsg (AnyMessage ps)
 
+-- requires UndecidableInstances extension
+instance Show (AnyMessage ps) => Show (TraceSendRecv ps) where
+  show (TraceSendMsg msg) = "Send " ++ show msg
+  show (TraceRecvMsg msg) = "Recv " ++ show msg
+
+-- |
+-- Tracer for @TraceSendRecv@.
+sendRecvDebugTracer :: (MonadSay m, Show (AnyMessage ps))
+                    => Tracer m (TraceSendRecv ps)
+sendRecvDebugTracer = Tracer (say . show)
```

Comment:
But I don't think we should include this one here. I don't think we should have core code depending on `MonadSay`. And it's easy enough to use at the call sites if it's needed.